### PR TITLE
[MaxiVault]Add Trisolaris MaxiVault

### PR DESCRIPTION
**Pull request by merlincdj**

### CODE REVIEW

File: contracts/earn/maxi/Trisolaris/MaxiVault.sol [hunk 1]
```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
```
Comment (merlincdj) [inline contracts/earn/maxi/Trisolaris/MaxiVault.sol:3]: `pragma experimental ABIEncoderV2;` will get compile error when using specific solidity feature, not need to using here.

File: contracts/earn/maxi/BaseMaxiVault.sol [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        depositAll();
+    }
+
+    function inCaseTokensGetStuck(address _stuckToken) external onlyAdmin {
+        require(_stuckToken != address(0), 'BaseMaxiVault: _stuckToken_ZERO_ADDRESS');
+        require(_stuckToken != stakingToken, 'BaseMaxiVault: _stuckToken_MUST_NOT_stakingToken');
+        require(_stuckToken != rewardToken, 'BaseMaxiVault: _stuckToken_MUST_NOT_rewardToken');
+
+        uint256 bal = IERC20(_stuckToken).balanceOf(address(this));
+        if (bal > 0) TransferHelper.safeTransfer(_stuckToken, msg.sender, bal);
```
Comment (fb-alexcq) [inline contracts/earn/maxi/BaseMaxiVault.sol:162]: format.

File: contracts/earn/maxi/BaseMaxiVault.sol [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    function inCaseNativeTokensGetStuck() external onlyAdmin {
+        // NOTE: vault never needs native tokens to do the yield farming;
+        // This native token balance indicates a user's incorrect transfer.
+        if (address(this).balance > 0) TransferHelper.safeTransferETH(msg.sender, address(this).balance);
```
Comment (fb-alexcq) [inline contracts/earn/maxi/BaseMaxiVault.sol:168]: format

File: contracts/earn/maxi/BaseMaxiVault.sol [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    modifier onlyAdmin() {
+        require(owner() == msg.sender || IWooAccessManager(wooAccessManager).isVaultAdmin(msg.sender), 'WOOFiMaximizerVault: NOT_ADMIN');
+        _;
+    }
+
+    /* ----- Internal Functions ----- */
+
+    function _chargePerformanceFee(address _token, uint256 _amount) internal returns (uint256) {
+        uint256 fee = _amount.mul(performanceFee).div(FEE_DENOMINATOR);
+        if (fee > 0) TransferHelper.safeTransfer(_token, performanceTreasury, fee);
```
Comment (fb-alexcq) [inline contracts/earn/maxi/BaseMaxiVault.sol:95]: format.

File: contracts/earn/maxi/Trisolaris/MaxiVault.sol [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        stakingToken = _stakingToken; // LP Token
+        rewardToken = _rewardToken;   // TRI
+
+        _giveAllowances();
+    }
+
+    /* ----- Modifier ----- */
+
+    modifier fairUpdate() {
+        uint256 bal = balance();
```
Comment (fb-alexcq) [inline contracts/earn/maxi/Trisolaris/MaxiVault.sol:124]: If this case, most of the time is balance is non-zero, just call harvest() then.

Comment (merlincdj) [reply contracts/earn/maxi/Trisolaris/MaxiVault.sol]: Agree, update code already.

File: contracts/earn/maxi/BaseMaxiVault.sol [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+import '../../interfaces/IWooAccessManager.sol';
+
+import '../../libraries/PausableUpgradeable.sol';
+
+abstract contract BaseMaxiVault is PausableUpgradeable, ReentrancyGuardUpgradeable {
+    using SafeMathUpgradeable for uint256;
+
+    /* ----- State Variables ----- */
+
+    address public stakingToken;
```
Comment (fb-alexcq) [inline contracts/earn/maxi/BaseMaxiVault.sol:54]: want

**Review by fb-alexcq — APPROVED** [review]

**Pull request merged**